### DISCUSSION TITLE
（已解决）开了443端口，输入域名后也没办法直接用https打开，怎么解决？

### DISCUSSION THREAD

**Discussion by Aceak** [AlistGo/alist#672] [Q&A]
配置文件里面把端口设置成443了，也打开了https，也添加了证书，然后在浏览器输入域名后却还是会默认以http://开头，并打不开网站，需要手动输入https://后才能打开，这个有办法让他直接打开https吗？
（用的一键安装脚本）

**Comment by xhofe** [comment]
可以说一下解决方法，供后续的使用者参考。

**Reply by Aceak** [reply]
我用的是nginx，我先反向代理把 `5244` 端口转发到了 `433` 端口，之后又把 `80` 端口也转发到了 `433` 端口，这样在浏览器地址栏里只输入域名，也能自动重定向到 `443` 端口，即使输入时用了 `http://` 也会自动重定向到 `https://`
相关配置：
```
server {
    listen 80;
    server_name YOUR_DOMAIN;
    rewrite ^(.*)$ https://${server_name}$1 permanent;     
}
```

**Reply by LoveGlaze** [reply]
alist配置端口443，开启https，安装nginx，配置文件贴上楼上的代码就可以了

**Comment by anwen-anyi** [comment]
QAQ 强制HTTPS不可以吗? ~

**Reply by Aceak** [reply]
alist配置完后应该需要配置一下nginx，我在alist里面开https后效果不太理想，所以就再用nginx转发了一下80端口

**Reply by LoveGlaze** [reply]
> alist配置完后应该需要配置一下nginx，我在alist里面开https后效果不太理想，所以就再用nginx转发了一下80端口

alist开启https改端口443需要手动输https才能访问，用nginx会吧访问80端口的http重定向为https，应该是这样吧，业余选手！

**Reply by Aceak** [reply]
是的
